From b49e8ccac86360ee5c2c085d20ffdb9e88c7b4f8 Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Thu, 9 Aug 2018 23:55:31 -0700
Subject: [PATCH 05/31] Iterate a CXXRecordDecl only if it has definition

---
 clang_delta/Transformation.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/clang_delta/Transformation.cpp b/clang_delta/Transformation.cpp
index a9d3b5e..29a7c33 100644
--- a/clang_delta/Transformation.cpp
+++ b/clang_delta/Transformation.cpp
@@ -510,6 +510,9 @@ const FunctionDecl *Transformation::lookupFunctionDeclFromBases(
         const CXXRecordDecl *CXXRD,
         DeclContextSet &VisitedCtxs)
 {
+  if (!CXXRD->hasDefinition()) {
+    return NULL;
+  }
   for (CXXRecordDecl::base_class_const_iterator I =
        CXXRD->bases_begin(), E = CXXRD->bases_end(); I != E; ++I) {
 
-- 
2.21.0

