From 0a4413089d7088f50ed8bbde74473aaedd284722 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Wed, 22 Aug 2018 10:29:48 +0000
Subject: [PATCH 06/31] Check number of CPUs only if --n wasn't specified

---
 creduce/creduce.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/creduce/creduce.in b/creduce/creduce.in
index 07f55c8..1586f93 100644
--- a/creduce/creduce.in
+++ b/creduce/creduce.in
@@ -45,7 +45,7 @@ if ($^O eq "MSWin32") {
 use creduce_config qw(PACKAGE_STRING);
 use creduce_utils;
 
-my $NPROCS = nprocs();
+my $NPROCS;
 
 ######################################################################
 
@@ -208,6 +208,7 @@ Getopt::Tabular::SetOptionPatterns qw|(--)([\w-]+) (-)(\w+)|;
 Getopt::Tabular::SetHelpOption("--help");
 GetOptions(\@options, \@ARGV) or exit(1);
 usage() unless (@ARGV >= 2);
+defined $NPROCS or $NPROCS = nprocs();
 
 my @custom_methods;
 
-- 
2.21.0

