From 78e434bb72f749a002dadb0390163210e63250e6 Mon Sep 17 00:00:00 2001
From: Jakub Wilk <jwilk@jwilk.net>
Date: Wed, 22 Aug 2018 12:34:02 +0200
Subject: [PATCH 07/31] Fix typo

---
 creduce/creduce.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/creduce/creduce.in b/creduce/creduce.in
index 07f55c8..c134cbb 100644
--- a/creduce/creduce.in
+++ b/creduce/creduce.in
@@ -134,7 +134,7 @@ timeout is that when a test is killed due to timing out, a message
 will be printed. If you do implement a timeout yourself, we recommend
 that your interestingness test returns exit code 124 when a subcommand
 times out. This will not affect the reduction but it will cause
-C-Reduce to print a message stating the the interestingness test
+C-Reduce to print a message stating the interestingness test
 reported a timeout. It is important to ensure that timeouts are not
 occurring too often (especially during C-Reduce's initial passes) or
 else C-Reduce will work poorly.
-- 
2.21.0

