From f0312d47bdc6ba9b1fb989832d8c8bc69c5b0fec Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Mon, 27 Aug 2018 22:53:15 -0700
Subject: [PATCH 10/31] Allow tests with .cc postfix

---
 clang_delta/tests/lit.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clang_delta/tests/lit.cfg b/clang_delta/tests/lit.cfg
index 0db613f..215258a 100644
--- a/clang_delta/tests/lit.cfg
+++ b/clang_delta/tests/lit.cfg
@@ -13,7 +13,7 @@ import os
 import sys
 
 config.name = 'clang_delta'
-config.suffixes = ['.c', '.cpp']
+config.suffixes = ['.c', '.cc', '.cpp']
 config.test_exec_root = config.builddir + '/tests'
 config.test_format = lit.formats.ShTest()
 config.test_source_root = os.path.dirname(__file__)
-- 
2.21.0

