From 9aa30eb7a804a39d04311a07c8b0e807fe542fa1 Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Tue, 28 Aug 2018 00:42:41 -0700
Subject: [PATCH 14/31] Fixed an issue for empty-struct-to-int

---
 clang_delta/EmptyStructToInt.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/clang_delta/EmptyStructToInt.cpp b/clang_delta/EmptyStructToInt.cpp
index 29615af..ae15c8a 100644
--- a/clang_delta/EmptyStructToInt.cpp
+++ b/clang_delta/EmptyStructToInt.cpp
@@ -449,6 +449,10 @@ bool EmptyStructToInt::isValidRecordDecl(const RecordDecl *RD)
         if (FD->hasBody() && !FD->isInlined())
           return false;
       }
+      if (const FieldDecl *FieldD = dyn_cast<FieldDecl>(*I)) {
+        if (pointToSelf(FieldD))
+          return false;
+      }
       ++count;
     }
   }
-- 
2.21.0

