From 8fc6028e654fd75b048c05190a9e326a6af22b2d Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Mon, 1 Oct 2018 20:35:53 -0700
Subject: [PATCH 17/31] Fixed a -Wreturn-type warning

---
 clang_delta/RewriteUtils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clang_delta/RewriteUtils.cpp b/clang_delta/RewriteUtils.cpp
index 895cf89..1adde91 100644
--- a/clang_delta/RewriteUtils.cpp
+++ b/clang_delta/RewriteUtils.cpp
@@ -1039,6 +1039,7 @@ bool RewriteUtils::replaceRecordDeclDef(const RecordDecl *RD,
       return !TheRewriter->ReplaceText(SourceRange(StartLoc, RBLoc), NameStr);
     }
   }
+  return true;
 }
 
 bool RewriteUtils::replaceVarTypeName(const VarDecl *VD,
-- 
2.21.0

