From 7959cbe101e46dd36cd7d14118a8c7a20c575b9c Mon Sep 17 00:00:00 2001
From: Amy Huang <akhuang@google.com>
Date: Tue, 19 Feb 2019 17:11:12 -0800
Subject: [PATCH 24/31] change ordering so pass_includes is called after
 pass_unifdef

---
 creduce/creduce.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/creduce/creduce.in b/creduce/creduce.in
index 52f9e6d..cd2aba5 100644
--- a/creduce/creduce.in
+++ b/creduce/creduce.in
@@ -755,12 +755,12 @@ sub line_delta_pass ($) {
 my @all_methods = (
 
     { "name" => "pass_include_includes", "arg" => "0",               "pri" => 100, "C" => 1, },
-    { "name" => "pass_includes", "arg" => "0",                                      "first_pass_pri" =>  0, "C" => 1, },
     { "name" => "pass_unifdef",  "arg" => "0",                       "pri" => 450,  "first_pass_pri" =>  0, "C" => 1, },
     { "name" => "pass_comments", "arg" => "0",                       "pri" => 451,  "first_pass_pri" =>  0, "C" => 1, },
-    { "name" => "pass_blank",    "arg" => "0",                                      "first_pass_pri" =>  1, },
-    { "name" => "pass_clang_binsrch",    "arg" => "replace-function-def-with-decl", "first_pass_pri" =>  2, "C" => 1, },
-    { "name" => "pass_clang_binsrch",    "arg" => "remove-unused-function",         "first_pass_pri" =>  3, "C" => 1, },
+    { "name" => "pass_includes", "arg" => "0",                                      "first_pass_pri" =>  1, "C" => 1, },
+    { "name" => "pass_blank",    "arg" => "0",                                      "first_pass_pri" =>  2, },
+    { "name" => "pass_clang_binsrch",    "arg" => "replace-function-def-with-decl", "first_pass_pri" =>  3, "C" => 1, },
+    { "name" => "pass_clang_binsrch",    "arg" => "remove-unused-function",         "first_pass_pri" =>  4, "C" => 1, },
 
     { "name" => "pass_lines",    "arg" => "0",                      "pri" => 410,  "first_pass_pri" =>  20,   "last_pass_pri" => 999, },
     { "name" => "pass_lines",    "arg" => "1",                      "pri" => 411,  "first_pass_pri" =>  21, },
-- 
2.21.0

