From e18bb56a4eb94865bc47f766a75e113b6aaf649d Mon Sep 17 00:00:00 2001
From: Yang Chen <chenyang@cs.utah.edu>
Date: Sun, 24 Feb 2019 19:20:02 -0800
Subject: [PATCH 30/31] skip inlining calls in default arguments

---
 clang_delta/SimpleInliner.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/clang_delta/SimpleInliner.cpp b/clang_delta/SimpleInliner.cpp
index df384bb..8728dcc 100644
--- a/clang_delta/SimpleInliner.cpp
+++ b/clang_delta/SimpleInliner.cpp
@@ -69,6 +69,8 @@ public:
 
   bool TraverseConstructorInitializer(CXXCtorInitializer *Init);
 
+  bool TraverseParmVarDecl(ParmVarDecl *PV);
+
   unsigned int getNumStmts(void) {
     return NumStmts;
   }
@@ -186,6 +188,11 @@ bool SimpleInlinerCollectionVisitor::TraverseConstructorInitializer(
   return true;
 }
 
+bool SimpleInlinerCollectionVisitor::TraverseParmVarDecl(ParmVarDecl *PV)
+{
+  return true;
+}
+
 bool SimpleInlinerFunctionVisitor::VisitReturnStmt(ReturnStmt *RS)
 {
   ConsumerInstance->ReturnStmts.push_back(RS);
-- 
2.21.0

