--- creduce-2.6.0/clang_delta/ExpressionDetector.cpp.omv~	2017-02-01 18:28:08.217081071 +0100
+++ creduce-2.6.0/clang_delta/ExpressionDetector.cpp	2017-02-01 18:28:33.111245632 +0100
@@ -452,8 +452,8 @@ bool ExpressionDetector::isIdenticalExpr
   }
 
   case Stmt::StringLiteralClass: {
-    const StringLiteral *Lit1 = cast<StringLiteral>(E1);
-    const StringLiteral *Lit2 = cast<StringLiteral>(E2);
+    const clang::StringLiteral *Lit1 = cast<clang::StringLiteral>(E1);
+    const clang::StringLiteral *Lit2 = cast<clang::StringLiteral>(E2);
     return Lit1->getBytes() == Lit2->getBytes();
   }
 
