diff -up creduce-llvm-12.0-dev/clang_delta/SimpleInliner.cpp.omv~ creduce-llvm-12.0-dev/clang_delta/SimpleInliner.cpp
--- creduce-llvm-12.0-dev/clang_delta/SimpleInliner.cpp.omv~	2021-05-05 16:21:05.110169057 +0200
+++ creduce-llvm-12.0-dev/clang_delta/SimpleInliner.cpp	2021-05-05 16:23:34.977945495 +0200
@@ -20,6 +20,7 @@
 #include "clang/AST/ASTContext.h"
 #include "clang/AST/Mangle.h"
 #include "clang/Basic/SourceManager.h"
+#include "clang/Basic/TargetInfo.h"
 
 #include "TransformationManager.h"
 #include "CommonStatementVisitor.h"
@@ -532,7 +533,7 @@ FunctionDecl *SimpleInliner::getAliaseeF
   if (const Attr *A = FD->getDefiningAttr()) {
     const AliasAttr *Alias = cast<AliasAttr>(A);
     llvm::StringRef Aliasee = Alias->getAliasee();
-    auto I = MangledNameToFuncDeclMap.find(Aliasee);
+    auto I = MangledNameToFuncDeclMap.find(Aliasee.str());
     if (I == MangledNameToFuncDeclMap.end())
       return nullptr;
     else
